tests/testthat/test-filter.R

```diff
@@ -0,0 +1,100 @@
+# TODO: implement these functions in filters.R and use them in this test
+include <- function(...) {}
+exclude <- function(...) {}
+lessthan <- function(...) {}
+morethan <- function(...) {}
+interval <- function(...) {}
+
+# TODO: implement filters, check number of samples after filtering the test data and fill expected_samples
+test_cases <- list(
+  list(
+    method = "by.meta", query = list(Status = include("C")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", method = "by.meta", query = list(Lane = exclude("A")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", query = list(Age = lessthan(23)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", query = list(Age = interval(15, 23)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", query = list(Lane = include("B", "C")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", query = list(Lane = exclude("A", "C")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.meta", query = list(Lane = include("A", "B"), Age = morethan(15)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.repertoire", query = list(n_clonotypes = morethan(1000)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.repertoire", query = list(n_clones = lessthan(100)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.rep", query = list(n_clonotypes = morethan(1000)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.rep", query = list(n_clones = lessthan(100)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.clonotype", query = list(CDR3.aa = exclude("partial", "out_of_frame")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.clonotype", query = list(Clones = interval(5, 50)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.cl", query = list(CDR3.aa = exclude("partial", "out_of_frame")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.cl", query = list(Clones = interval(5, 50)),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")),
+    expected_samples = 0
+  ),
+  list(
+    method = "by.gene", query = list(V = include("TRAV9"), J = include("TRAJ11")),
+    expected_samples = 0
+  )
+)
+
+for (i in seq_along(test_cases)) {
```

Comment:
I don't see the arrange, act, assert structure

Reply:
> 
> 
> Do you always expect that the result will return zero samples?

No, test dataset need to be filtered to check how many samples will there actually be; or I can take small dataset for the test and count values manually

Reply:
> 
> 
> I don't see the arrange, act, assert structure

It is here; the loop is analogous to parametrized tests in Python. I will add the comments where are these parts.